specreduce/tracing.py

```diff
@@ -221,17 +226,23 @@ def __post_init__(self):
         if self._disp_axis != 1:
             raise ValueError('dispersion axis must equal 1')
 
+        cols = img.shape[self._disp_axis]
+        if self.bins < 4:
+            # many of the Astropy model fitters require four points at minimum
+            raise ValueError('bins must be >= 4')
+        elif self.bins > cols:
+            raise ValueError(f"bins must be <= {cols}, the length of the "
+                             "image's spatial direction")
+
         if not isinstance(self.bins, int):
             warnings.warn('TRACE: Converting bins to int')
             self.bins = int(self.bins)
 
-        if self.bins < 4:
-            raise ValueError('bins must be >= 4')
-
-        cols = img.shape[self._disp_axis]
-        if self.bins >= cols:
-            raise ValueError(f"bins must be < {cols}, the length of the "
-                             "image's spatial direction")
+        valid_models = ["Spline1D", "Legendre1D", "Chebyshev1D", "Polynomial1D"]
+        trace_model_name = type(self.trace_model).name
+        if trace_model_name not in valid_models:
+            raise ValueError("trace_model must be one of "
+                             f"{', '.join(valid_models)}.")
```

Comment:
is there a benefit to this over isinstance checks?  In theory, this would let some non-astropy model object with the same class name through the check

Reply:
Although the non-spline options are `models.PolynomialModel` instances, so are many other Astropy models. Their most common ancestor with `Spline1D` is `astropy.modeling.core.FittableModel`, so maybe I could check for that and then check that the name of the type comes from the `valid_models` list I've already defined.

Reply:
since you need a list somewhere anyways (unless you want to open it to any generic model), I think its just as short to do:

```
from astropy.modeling import models

valid_models = (models.Spline1D, models.Legendre1D, models.Chebyshev1D, models.Polynomial1D)
if not isinstance(trace_model, valid_models):
    raise ValueError(...)
```